python/utils/transactions_processor.py

```diff
@@ -0,0 +1,140 @@
+        self.engine = create_engine(self.config.db_connection_uri)
+
+        # Check if the tables are created.
+        inspector = inspect(self.engine)
+        table_name = NextVersionToProcess.__tablename__
+        if inspector.has_table(table_name):
+            print("Table {} exists.".format(table_name))
+        else:
+            print("Table {} does not exist. Creating it now.".format(table_name))
+            Base.metadata.create_all(self.engine)
+
+    def process(self) -> None:
+        # Setup the GetTransactionsRequest
+        starting_version = self.config.get_starting_version()
+        request = raw_data_pb2.GetTransactionsRequest(starting_version=starting_version)
+
+        # Setup GRPC settings
+        metadata = (
+            ("x-aptos-data-authorization", self.config.indexer_api_key),
+            ("x-aptos-request-name", self.config.indexer_name),
+        )
+        options = [("grpc.max_receive_message_length", -1)]
+
+        print(
+            json.dumps(
+                {
+                    "message": "Connected to the indexer grpc",
+                    "starting_version": starting_version,
+                }
+            )
+        )
+
+        # Connect to indexer grpc endpoint
+        with grpc.insecure_channel(
+            self.config.indexer_endpoint, options=options
+        ) as channel:
+            stub = raw_data_pb2_grpc.RawDataStub(channel)
+            current_transaction_version = starting_version
+
+            for response in stub.GetTransactions(
+                request,
+                metadata=metadata,
+            ):
+                chain_id = response.chain_id
+
+                if chain_id != self.config.chain_id:
+                    raise Exception(
+                        "Chain ID mismatch. Expected chain ID is: "
+                        + str(self.config.chain_id)
+                        + ", but received chain ID is: "
+                        + str(chain_id)
+                    )
+                print(
+                    json.dumps(
+                        {
+                            "message": "Response received",
+                            "starting_version": response.transactions[0].version,
+                        }
+                    )
+                )
+
+                transactions_output = response
+                for transaction in transactions_output.transactions:
+                    transaction_version = transaction.version
+                    if transaction_version != current_transaction_version:
+                        raise Exception(
+                            "Transaction version mismatch. Expected transaction version is: "
+                            + str(current_transaction_version)
+                            + ", but received transaction version is: "
+                            + str(transaction_version)
+                        )
+
+                    parsed_objs = self.parser_function(transaction)
```

Comment:
what's the plan for `pip` pkg management and how to use the library? what they can import and what they need to build themself?

Reply:
I think the idea with this is they can import this `TransactionsProcessor` and they have to implement `parser_function` and define the DB schema themselves.